### PR TITLE
Erika_Smith_1708_Scrabble

### PR DESCRIPTION
**Pull request by erikaannesmith**

### CODE REVIEW

File: lib/scrabble.rb [hunk 1]
```diff
@@ -15,4 +17,24 @@ def point_values
...
     }
   end
+
+  def score(word)
+    if word.nil?
+      @score
+    else
+      letters = word.upcase.split("")
+
+      letters.each do |letter|
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:27]: Would love to see you push to use something like reduce here or map & sum instead of each.

File: lib/scrabble.rb [hunk 2]
```diff
@@ -15,4 +17,24 @@ def point_values
...
+        @score += point_values[letter]
+      end
+    @score
+    end
+  end
+
+  def score_with_multipliers(word, letter_multiplier, word_multiplier =1)
+    letters = word.upcase.split("")
+
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:36]: Watch out for extra spaces (I recommend giving your code a quick glance before you put in a final product to clean up things like this)

File: test/scrabble_test.rb [hunk 3]
```diff
@@ -6,8 +6,21 @@
 
 class ScrabbleTest < Minitest::Test
   def test_it_can_score_a_single_letter
-    skip
     assert_equal 1, Scrabble.new.score("a")
+
     assert_equal 4, Scrabble.new.score("f")
   end
+
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:13]: Make sure to add tests for every piece of functionality (you don't have any for whole words).

File: test/scrabble_test.rb [hunk 4]
```diff
@@ -6,8 +6,21 @@
...
+  def test_it_can_score_an_empty_word
```
Comment (VictoriaVasys) [inline test/scrabble_test.rb:14]: I like that you broke out separate tests for the edge cases!

File: lib/scrabble.rb [hunk 5]
```diff
@@ -15,4 +17,24 @@ def point_values
       "Y"=>4, "Z"=>10
     }
   end
+
+  def score(word)
```
Comment (VictoriaVasys) [inline lib/scrabble.rb:21]: This method is a little long. Think about ways you could break out logic.

**Review by VictoriaVasys — COMMENTED** [review]
Overall, good job Erika. See my comments for specifics, but I'd love to see you pushing to create more tests for robust functionality & dealing with edge cases & breaking out logic a bit more. Also make sure you're committing about every 15-20 minutes.
